mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -2232,7 +2239,7 @@ private void resubmitWorker(JobWorker oldWorker) throws Exception {
             Map<Integer, Integer> workerToStageMap = mantisJobMetaData.getWorkerNumberToStageMap();
 
             IMantisWorkerMetadata oldWorkerMetadata = oldWorker.getMetadata();
-            if (oldWorkerMetadata.getTotalResubmitCount()
+            if (recentlyResubmittedWorkersCache.size()
```

Comment:
From the [doc](https://guava.dev/releases/15.0/api/docs/com/google/common/cache/CacheBuilder.html#expireAfterWrite(long,%20java.util.concurrent.TimeUnit)) size() could be inaccurate, what about trigger cleanup first?